app/controllers/divisions_controller.rb

```diff
@@ -8,29 +8,34 @@ def index_redirect
   end
 
   def index
+    @years = (Division.order(:date).first.date.year..Division.order(:date).last.date.year).to_a
+
     if params[:mpc] && params[:mpn]
-      electorate = params[:mpc].gsub("_", " ")
-      name = params[:mpn].gsub("_", " ")
+      @mpc = params[:mpc]
+      @mpn = params[:mpn]
+      @house = params[:house]
 
-      @member = Member.with_name(name)
-      @member = @member.in_house(params[:house])
-      @member = @member.where(constituency: electorate)
-      @member = @member.order(entered_house: :desc).first
+      @member = get_member
 
-      if @member.nil?
-        render 'members/member_not_found', status: 404
-      else
+      begin
+        @date_start, @date_end = get_date_range(params[:date])
+      rescue ArgumentError
+        render 'home/error_404', status: 404
+      end
+
+      if @member
+        @divisions = @member.divisions_possible(@date_start, @date_end).order(date: :desc, clock_time: :desc, name: :asc)
         render 'index_with_member'
+      else
+        render 'members/member_not_found', status: 404
       end
     else
-      @years = (Division.order(:date).first.date.year..Division.order(:date).last.date.year).to_a
       @sort = params[:sort]
       @rdisplay = params[:rdisplay]
       @house = params[:house] unless params[:house] == "all"
 
       begin
-        @date_start, @date_end, @date_range = DivisionParameterParser.get_date_range(params[:date]) if params[:date]
-        @date_start, @date_end, @date_range = DivisionParameterParser.get_date_range(@years.last.to_s) if @date_start.nil? && @date_end.nil? && @rdisplay.nil?
+        @date_start, @date_end, @date_range = get_date_range(params[:date], @rdisplay)
```

Comment:
I am not sure what this `@rdisplay` does. But it seems something is wrong with this field. The `Parliament` class also looks pretty strange. Are we using this at all in production? Is that important or can be removed?